### PR TITLE
JLab extension: render the preview on save

### PR DESCRIPTION
**Pull request by jtpio**
### Changes

- Add `plugin.json` to store the settings for the voila preview extension.
- Add the `renderOnSave` option to render the preview when the notebook is saved. Defaults to `false`.

image

### Follow-up

- Add a "Render on Save" checkbox in the toolbar?
- Make the voila preview panel a document widget extension?

### CODE REVIEW

**Comment by jtpio** [comment]
Added the "Render on Save" checkbox to the toolbar:

jlab-render-on-save

File: packages/jupyterlab-voila/src/preview.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            this.renderOnSave = event.target.checked;
+          }}
+        />
+        Render on Save
+      </label>
+    );
+    renderOnSaveCheckbox.addClass("jp-VoilaPreview-renderOnSave");
+
+    this.toolbar.addItem("reload", reloadButton);
+    this.toolbar.addItem("renderOnSave", renderOnSaveCheckbox);
```
Comment (maartenbreddels) [inline packages/jupyterlab-voila/src/preview.tsx]: jlab/phosphorjs supports react components?

**Comment by maartenbreddels** [comment]
LGTM, do you wanna squash and merge?

File: packages/jupyterlab-voila/src/preview.tsx (hunk 1 above)
Comment (jtpio) [reply packages/jupyterlab-voila/src/preview.tsx]: Yes (for jlab), most of the toolbar components are actually built with React (using the `ReactWidget` utility under the hood)

**Comment by jtpio** [comment]
I think the commits are relatively small and self-contained for a normal merge (without squashing / losing history)?

**Pull request merged**